Filter - from query to filter

PatrickCQ opened the pull request:
describe filtering on tree

PatrickCQ committed:
rename to filtering and add intro

PatrickCQ committed:
add filter idea and example

PatrickCQ committed:
add rule descriptions

PatrickCQ committed:
add rule false

PatrickCQ committed:
clarify rules return

PatrickCQ committed:
add combination of rules and rule example (not)

PatrickCQ committed:
add dfn and dfn-links

PatrickCQ committed:
add and and or rule

PatrickCQ committed:
add additional rules

PatrickCQ committed:
small cleanup

PatrickCQ committed:
fix dfn link

PatrickCQ committed:
remove node description

PatrickCQ committed:
rework rules

UlfBj commented:
It seems the filter model described here is abandoning the wildcard mechanism. It may be ok, but then wildcard expressions should not be used in the description of the model. 
Are the query rules expressed here, and its syntax, used elsewhere before?
If not, we need to scrutinize it very carefully before bringing it aboard. 
There is an ongoing discussion at VSS to abandon wildcards, together with introducing changes to the model and tools. Maybe we should wait for that to settle before concluding here?

PeterWinzell reviewed (commented):
1) "_Most times it is not the best way to get a whole part of the data representation from the service when looking for a specific set with a specific value"_. This section describes how to build the queries to search for specific parts of the data representation.

**The first sentence is a bit long and i actually think it does not add any real value. I would skip and just use "This section..."**

2)
 ...is an expression that describes a characteristic of nodes that shall be returned by a <a>query</a>.
			<a>Rules</a> are applied to a leaf and **descibe** a desired attribute of a node...

typo: should be describe.

PatrickCQ committed:
update typo

PatrickCQ committed:
remove intention description

the description for the intention of the change is only valid in the development of this specification and does not add to the value of the specifciation.

PatrickCQ committed:
Merge branch 'gh-pages-w3c' into filter

danielwilms commented on spec/Gen2_Core.html:
```diff
@@ -298,51 +298,175 @@ <h2>Service discovery</h2>
+		<h3>Example</h3>
+			<p>
+			All open doors shall be given. The tree looks as follows:
+			<blockquote>
```
will be outdated in V2

danielwilms commented on spec/Gen2_Core.html:
```diff
@@ -298,51 +298,175 @@ <h2>Service discovery</h2>
+			<p>
+			The <a>path</a> that shall be inspected is "/vehicle/cabin". <br>
+			The <a>rule</a> is that "/vehicle/cabin/*/*/Door/isLocked" shall be "true".<br>
+			</p>
```
Consider https://github.com/GENIVI/vehicle_signal_specification/issues/81 as input. would already take that one as base.
